src/http_server.cc

```diff
@@ -2765,6 +2765,38 @@ HTTPAPIServer::EVBufferToRawInput(
   return nullptr;  // success
 }
 
+struct HeaderSearchPayload {
+  const char* regex_;
+  TRITONSERVER_InferenceRequest* request_;
+  TRITONSERVER_Error* error_;
+};
+
+int
+ForEachHeader(evhtp_header_t* header, void* arg)
+{
+  HeaderSearchPayload* header_search_payload =
+      reinterpret_cast<HeaderSearchPayload*>(arg);
+
+  TRITONSERVER_InferenceRequest* irequest = header_search_payload->request_;
+  const char* pattern = header_search_payload->regex_;
+
+  std::regex base_regex(pattern);
+  if (std::regex_search(header->key, base_regex)) {
```

Comment:
As I understand it, this will do regex matching on every header in every request. I imagine this could be quite expensive on the critical path. 

Have you done any kind of ballpark benchmarking on what latency this adds? If the latency is significant, would something like a `string.startsWith` be better(and enforcing some prefix/format for forwarded headers)?

Reply:
It is for pre-compiled regex.

Reply:
It is for pre-compiled regex.

Reply:
What's the cost of forwarding all headers to parameters? Is there a need to only forward specific ones?

Reply:
Depends on the number of headers and their values. For Python backend, we have to a copy of all of them to pass it through shared memory.

Reply:
Changed to prefix string matching 👍